File: base-class-module/src/main/java/jingya/com/base_class_module/BaseMvps/BaseMvpPresenter.java
package jingya.com.base_class_module.BaseMvps;

/**
 * @author kuky
 * @description presenter 基类
 */
public abstract class BaseMvpPresenter<V extends BaseMvpViewImpl> {
    protected V mView;

    protected void attach(V view) {
        this.mView = view;
    }

    protected void detach() {
        this.mView = null;
    }

    protected void onResume() {
    }

    protected void onStop() {
    }

    protected void onPause() {
    }
}

File: base-kt-module/src/main/java/com/kuky/base_kt_module/BaseMvps/BaseMvpPresenter.kt
package com.kuky.base_kt_module.BaseMvps


/**
 * @author kuky
 * @description
 */
abstract class BaseMvpPresenter<in V : BaseMvpViewImpl> {
    private var mView: V? = null

    fun attachView(view: V) {
        this.mView = view
    }

    fun detachView() {
        this.mView = null
    }

    fun onResume() {

    }

    fun onStop() {

    }

    fun onPause() {

    }
}